API::PageConfiguration::copy() is error-prone

cdumez opened the pull request:
#### ecce6504d149b63d13a78cda3b352b240d2569c4
<pre>
API::PageConfiguration::copy() is error-prone
<a href="https://bugs.webkit.org/show_bug.cgi?id=257402">https://bugs.webkit.org/show_bug.cgi?id=257402</a>

Reviewed by Sihui Liu.

API::PageConfiguration::copy() is error-prone as it is too easy to add a data
member and forget to copy it in this function (Like in 264609@main).

Refactor the class so that API::PageConfiguration::copy() no longer needs
updating when adding new data members.

* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::copy const):
(API::PageConfiguration::processPool):
(API::PageConfiguration::setProcessPool):
(API::PageConfiguration::userContentController):
(API::PageConfiguration::setUserContentController):
(API::PageConfiguration::webExtensionController):
(API::PageConfiguration::setWebExtensionController):
(API::PageConfiguration::weakWebExtensionController):
(API::PageConfiguration::setWeakWebExtensionController):
(API::PageConfiguration::pageGroup):
(API::PageConfiguration::setPageGroup):
(API::PageConfiguration::preferences):
(API::PageConfiguration::setPreferences):
(API::PageConfiguration::relatedPage const):
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::pageToCloneSessionStorageFrom const):
(API::PageConfiguration::setPageToCloneSessionStorageFrom):
(API::PageConfiguration::visitedLinkStore):
(API::PageConfiguration::setVisitedLinkStore):
(API::PageConfiguration::websiteDataStore):
(API::PageConfiguration::setWebsiteDataStore):
(API::PageConfiguration::defaultWebsitePolicies const):
(API::PageConfiguration::setDefaultWebsitePolicies):
(API::PageConfiguration::urlSchemeHandlerForURLScheme):
(API::PageConfiguration::setURLSchemeHandlerForURLScheme):
(API::PageConfiguration::lockdownModeEnabled const):
(API::PageConfiguration::setDelaysWebProcessLaunchUntilFirstLoad):
(API::PageConfiguration::delaysWebProcessLaunchUntilFirstLoad const):
(API::PageConfiguration::isLockdownModeExplicitlySet const):
(API::PageConfiguration::applicationManifest const):
(API::PageConfiguration::setApplicationManifest):
(API::PageConfiguration::PageConfiguration): Deleted.
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::clientNavigationsRunAtForegroundPriority const):
(API::PageConfiguration::setClientNavigationsRunAtForegroundPriority):
(API::PageConfiguration::canShowWhileLocked const):
(API::PageConfiguration::setCanShowWhileLocked):
(API::PageConfiguration::clickInteractionDriverForTesting const):
(API::PageConfiguration::setClickInteractionDriverForTesting):
(API::PageConfiguration::initialCapitalizationEnabled):
(API::PageConfiguration::setInitialCapitalizationEnabled):
(API::PageConfiguration::cpuLimit const):
(API::PageConfiguration::setCPULimit):
(API::PageConfiguration::waitsForPaintAfterViewDidMoveToWindow const):
(API::PageConfiguration::setWaitsForPaintAfterViewDidMoveToWindow):
(API::PageConfiguration::drawsBackground const):
(API::PageConfiguration::setDrawsBackground):
(API::PageConfiguration::isControlledByAutomation const):
(API::PageConfiguration::setControlledByAutomation):
(API::PageConfiguration::overrideContentSecurityPolicy const):
(API::PageConfiguration::setOverrideContentSecurityPolicy):
(API::PageConfiguration::additionalSupportedImageTypes const):
(API::PageConfiguration::setAdditionalSupportedImageTypes):
(API::PageConfiguration::urlSchemeHandlers):
(API::PageConfiguration::corsDisablingPatterns const):
(API::PageConfiguration::setCORSDisablingPatterns):
(API::PageConfiguration::maskedURLSchemes const):
(API::PageConfiguration::setMaskedURLSchemes):
(API::PageConfiguration::userScriptsShouldWaitUntilNotification const):
(API::PageConfiguration::setUserScriptsShouldWaitUntilNotification):
(API::PageConfiguration::crossOriginAccessControlCheckEnabled const):
(API::PageConfiguration::setCrossOriginAccessControlCheckEnabled):
(API::PageConfiguration::processDisplayName const):
(API::PageConfiguration::setProcessDisplayName):
(API::PageConfiguration::loadsSubresources const):
(API::PageConfiguration::setLoadsSubresources):
(API::PageConfiguration::allowedNetworkHosts const):
(API::PageConfiguration::setAllowedNetworkHosts):
(API::PageConfiguration::ignoresAppBoundDomains const):
(API::PageConfiguration::setIgnoresAppBoundDomains):
(API::PageConfiguration::limitsNavigationsToAppBoundDomains const):
(API::PageConfiguration::setLimitsNavigationsToAppBoundDomains):
(API::PageConfiguration::setMediaCaptureEnabled):
(API::PageConfiguration::mediaCaptureEnabled const):
(API::PageConfiguration::setHTTPSUpgradeEnabled):
(API::PageConfiguration::httpsUpgradeEnabled const):
(API::PageConfiguration::setShouldRelaxThirdPartyCookieBlocking):
(API::PageConfiguration::shouldRelaxThirdPartyCookieBlocking const):
(API::PageConfiguration::setAttributedBundleIdentifier):
(API::PageConfiguration::attributedBundleIdentifier const):
(API::PageConfiguration::appInitiatedOverrideValueForTesting const):
(API::PageConfiguration::setAppInitiatedOverrideValueForTesting):
(API::PageConfiguration::requiresUserActionForEditingControlsManager const):
(API::PageConfiguration::setRequiresUserActionForEditingControlsManager):
(API::PageConfiguration::setAllowTestOnlyIPC):
(API::PageConfiguration::allowTestOnlyIPC const):
(API::PageConfiguration::setContentSecurityPolicyModeForExtension):
(API::PageConfiguration::contentSecurityPolicyModeForExtension const):

Canonical link: <a href="https://commits.webkit.org/264689@main">https://commits.webkit.org/264689@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de96704f8180b9f0ca57e6e8bc0f3faf00850b3a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8346 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8639 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8856 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10014 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8411 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10627 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8550 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11277 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8492 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9548 "1 flakes 3 failures") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10160 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6846 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15199 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7967 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7780 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11128 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8251 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6726 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7540 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2015 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11750 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7994 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/UIProcess/API/APIPageConfiguration.h:
```diff
@@ -155,140 +155,140 @@ class PageConfiguration : public ObjectImpl<Object::Type::PageConfiguration> {
+        WebCore::ContentSecurityPolicyModeForExtension contentSecurityPolicyModeForExtension { WebCore::ContentSecurityPolicyModeForExtension::None };
+    };
+
+    Data m_data;
```
Could you add a comment that new member should be added to `Data` to avoid breaking copy?

szewai reviewed (approved):
r=me

cdumez committed:
API::PageConfiguration::copy() is error-prone
https://bugs.webkit.org/show_bug.cgi?id=257402

Reviewed by Sihui Liu.

API::PageConfiguration::copy() is error-prone as it is too easy to add a data
member and forget to copy it in this function (Like in 264609@main).

Refactor the class so that API::PageConfiguration::copy() no longer needs
updating when adding new data members.

* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::copy const):
(API::PageConfiguration::processPool):
(API::PageConfiguration::setProcessPool):
(API::PageConfiguration::userContentController):
(API::PageConfiguration::setUserContentController):
(API::PageConfiguration::webExtensionController):
(API::PageConfiguration::setWebExtensionController):
(API::PageConfiguration::weakWebExtensionController):
(API::PageConfiguration::setWeakWebExtensionController):
(API::PageConfiguration::pageGroup):
(API::PageConfiguration::setPageGroup):
(API::PageConfiguration::preferences):
(API::PageConfiguration::setPreferences):
(API::PageConfiguration::relatedPage const):
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::pageToCloneSessionStorageFrom const):
(API::PageConfiguration::setPageToCloneSessionStorageFrom):
(API::PageConfiguration::visitedLinkStore):
(API::PageConfiguration::setVisitedLinkStore):
(API::PageConfiguration::websiteDataStore):
(API::PageConfiguration::setWebsiteDataStore):
(API::PageConfiguration::defaultWebsitePolicies const):
(API::PageConfiguration::setDefaultWebsitePolicies):
(API::PageConfiguration::urlSchemeHandlerForURLScheme):
(API::PageConfiguration::setURLSchemeHandlerForURLScheme):
(API::PageConfiguration::lockdownModeEnabled const):
(API::PageConfiguration::setDelaysWebProcessLaunchUntilFirstLoad):
(API::PageConfiguration::delaysWebProcessLaunchUntilFirstLoad const):
(API::PageConfiguration::isLockdownModeExplicitlySet const):
(API::PageConfiguration::applicationManifest const):
(API::PageConfiguration::setApplicationManifest):
(API::PageConfiguration::PageConfiguration): Deleted.
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::clientNavigationsRunAtForegroundPriority const):
(API::PageConfiguration::setClientNavigationsRunAtForegroundPriority):
(API::PageConfiguration::canShowWhileLocked const):
(API::PageConfiguration::setCanShowWhileLocked):
(API::PageConfiguration::clickInteractionDriverForTesting const):
(API::PageConfiguration::setClickInteractionDriverForTesting):
(API::PageConfiguration::initialCapitalizationEnabled):
(API::PageConfiguration::setInitialCapitalizationEnabled):
(API::PageConfiguration::cpuLimit const):
(API::PageConfiguration::setCPULimit):
(API::PageConfiguration::waitsForPaintAfterViewDidMoveToWindow const):
(API::PageConfiguration::setWaitsForPaintAfterViewDidMoveToWindow):
(API::PageConfiguration::drawsBackground const):
(API::PageConfiguration::setDrawsBackground):
(API::PageConfiguration::isControlledByAutomation const):
(API::PageConfiguration::setControlledByAutomation):
(API::PageConfiguration::overrideContentSecurityPolicy const):
(API::PageConfiguration::setOverrideContentSecurityPolicy):
(API::PageConfiguration::additionalSupportedImageTypes const):
(API::PageConfiguration::setAdditionalSupportedImageTypes):
(API::PageConfiguration::urlSchemeHandlers):
(API::PageConfiguration::corsDisablingPatterns const):
(API::PageConfiguration::setCORSDisablingPatterns):
(API::PageConfiguration::maskedURLSchemes const):
(API::PageConfiguration::setMaskedURLSchemes):
(API::PageConfiguration::userScriptsShouldWaitUntilNotification const):
(API::PageConfiguration::setUserScriptsShouldWaitUntilNotification):
(API::PageConfiguration::crossOriginAccessControlCheckEnabled const):
(API::PageConfiguration::setCrossOriginAccessControlCheckEnabled):
(API::PageConfiguration::processDisplayName const):
(API::PageConfiguration::setProcessDisplayName):
(API::PageConfiguration::loadsSubresources const):
(API::PageConfiguration::setLoadsSubresources):
(API::PageConfiguration::allowedNetworkHosts const):
(API::PageConfiguration::setAllowedNetworkHosts):
(API::PageConfiguration::ignoresAppBoundDomains const):
(API::PageConfiguration::setIgnoresAppBoundDomains):
(API::PageConfiguration::limitsNavigationsToAppBoundDomains const):
(API::PageConfiguration::setLimitsNavigationsToAppBoundDomains):
(API::PageConfiguration::setMediaCaptureEnabled):
(API::PageConfiguration::mediaCaptureEnabled const):
(API::PageConfiguration::setHTTPSUpgradeEnabled):
(API::PageConfiguration::httpsUpgradeEnabled const):
(API::PageConfiguration::setShouldRelaxThirdPartyCookieBlocking):
(API::PageConfiguration::shouldRelaxThirdPartyCookieBlocking const):
(API::PageConfiguration::setAttributedBundleIdentifier):
(API::PageConfiguration::attributedBundleIdentifier const):
(API::PageConfiguration::appInitiatedOverrideValueForTesting const):
(API::PageConfiguration::setAppInitiatedOverrideValueForTesting):
(API::PageConfiguration::requiresUserActionForEditingControlsManager const):
(API::PageConfiguration::setRequiresUserActionForEditingControlsManager):
(API::PageConfiguration::setAllowTestOnlyIPC):
(API::PageConfiguration::allowTestOnlyIPC const):
(API::PageConfiguration::setContentSecurityPolicyModeForExtension):
(API::PageConfiguration::contentSecurityPolicyModeForExtension const):

Canonical link: https://commits.webkit.org/264689@main
